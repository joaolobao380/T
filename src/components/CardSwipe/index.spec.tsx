import { render } from '@testing-library/react-native';
import React from 'react';

import { CardSwipe } from '.';

jest.mock('react-native/Libraries/Alert/Alert', () => ({
  alert: jest.fn(),
}));

describe('CardSwipe Component', () => {
  it('renders correctly', () => {
    const { getByText } = render(<CardSwipe title="Test Card" status="on" />);
    expect(getByText('Test Card')).toBeTruthy();
    expect(getByText('Ligado')).toBeTruthy();
  });

  it('displays the correct status', () => {
    const { getByText, rerender } = render(<CardSwipe title="Test Card" status="on" />);
    expect(getByText('Ligado')).toBeTruthy();

    rerender(<CardSwipe title="Test Card" status="off" />);
    expect(getByText('Desligado')).toBeTruthy();
  });

  it('renders correctly', () => {
    const { toJSON } = render(<CardSwipe title="Test Card" status="on" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
