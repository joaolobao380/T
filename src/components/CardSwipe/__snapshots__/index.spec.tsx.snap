// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSwipe Component renders correctly 1`] = `
<View
  collapsable={false}
  handlerTag={5}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "backgroundColor": "#2A2A2A",
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EC4848",
          "borderBottomRightRadius": 8,
          "borderTopRightRadius": 8,
          "height": "100%",
          "justifyContent": "center",
          "marginLeft": -8,
          "opacity": 1,
          "width": 80,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGPath
            d="M3 6h18"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
            x1="10"
            x2="10"
            y1="11"
            y2="17"
          />
          <RNSVGLine
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
            x1="14"
            x2="14"
            y1="11"
            y2="17"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    handlerTag={6}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2A2A2A",
          "borderRadius": 8,
          "flexDirection": "row",
          "paddingHorizontal": 16,
          "paddingVertical": 24,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Poppins-Regular",
              "fontSize": 16,
            }
          }
        >
          Test Card
        </Text>
        <View
          style={
            {
              "backgroundColor": "#6EFE85",
              "borderRadius": 16,
              "paddingHorizontal": 12,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              {
                "fontFamily": "Poppins-Regular",
                "fontSize": 14,
              }
            }
          >
            Ligado
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
